### PR TITLE
Enable async queries on Cosmos SQL

### PR DESCRIPTION
**Pull request by AndriySvyryd**
Handle Take, Skip, Last pushdown for SelectAsync
Remove async-to-sync subquery conversion when async is expected

### CODE REVIEW

File: src/EFCore/Query/EntityQueryModelVisitor.cs [hunk 1]
```diff
@@ -1350,7 +1350,7 @@ var selector
                             _expression,
                             Expression.Lambda(selector, CurrentParameter))
                         : Expression.Call(
-                            SelectAsyncMethod
```
Comment (smitpatel) [inline src/EFCore/Query/EntityQueryModelVisitor.cs]: yay

File: src/EFCore.Cosmos.Sql/Storage/Internal/CosmosClient.cs [hunk 2]
```diff
@@ -42,5 +43,18 @@ public class CosmosClient
                     sqlQuerySpec)
                 .GetEnumerator();
         }
+
+        public IDocumentQuery<Document> ExecuteAsyncSqlQuery(
```
Comment (smitpatel) [inline src/EFCore.Cosmos.Sql/Storage/Internal/CosmosClient.cs]: ExecuteSqlQueryAsync?

File: src/EFCore.Cosmos.Sql/Storage/Internal/CosmosClient.cs [hunk 3]
```diff
@@ -42,5 +43,18 @@ public class CosmosClient
...
+            string collectionId,
+            SqlQuerySpec sqlQuerySpec)
+        {
+            _commandLogger.ExecutingSqlQuery(sqlQuerySpec);
+
+            return DocumentClient
+                .CreateDocumentQuery<Document>(
+                    UriFactory.CreateDocumentCollectionUri(DatabaseId, collectionId),
+                    sqlQuerySpec)
+                .AsDocumentQuery();
```
Comment (smitpatel) [inline src/EFCore.Cosmos.Sql/Storage/Internal/CosmosClient.cs]: Does this actually execute query async? Can it be unified with sync?

File: src/EFCore.Cosmos.Sql/Storage/Internal/CosmosClient.cs (hunk 2 above)
Comment (AndriySvyryd) [reply src/EFCore.Cosmos.Sql/Storage/Internal/CosmosClient.cs]: It doesn't return a `Task` or an `IAsyncEnumerable` ...

File: src/EFCore.Cosmos.Sql/Storage/Internal/CosmosClient.cs (hunk 3 above)
Comment (AndriySvyryd) [reply src/EFCore.Cosmos.Sql/Storage/Internal/CosmosClient.cs]: It returns a query that can only be executed asynchronously. The API is weird.

**Pull request merged**